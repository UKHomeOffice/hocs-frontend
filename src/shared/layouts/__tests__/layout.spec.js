import React from 'react';
import Layout from '../layout.jsx';

describe('Page layout component', () => {
    it('should render with default props', () => {
        expect(
            shallow(<Layout />)
        ).toMatchSnapshot();
    });
    it('should render with footer when set', () => {
        expect(
            shallow(<Layout footer={{ isVisible: true }} />)
        ).toMatchSnapshot();
    });
    it('should render with children when passed', () => {
        expect(
            shallow(<Layout>Inside my layout</Layout>)
        ).toMatchSnapshot();
    });
});