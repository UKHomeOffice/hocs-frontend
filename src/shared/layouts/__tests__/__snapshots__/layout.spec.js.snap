// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page layout component should render with children when passed 1`] = `
<React.Fragment>
  <Header
    logoLinkTitle=""
    menu={
      Array [
        Object {
          "label": "Logout",
          "target": "/",
        },
      ]
    }
    propositionHeader="Service Name"
    propositionHeaderLink="/"
    service="GOV.UK"
    serviceLink="https://www.gov.uk"
  />
  <Body
    phaseBanner={
      Object {
        "isVisible": false,
      }
    }
  >
    Inside my layout
  </Body>
</React.Fragment>
`;

exports[`Page layout component should render with default props 1`] = `
<React.Fragment>
  <Header
    logoLinkTitle=""
    menu={
      Array [
        Object {
          "label": "Logout",
          "target": "/",
        },
      ]
    }
    propositionHeader="Service Name"
    propositionHeaderLink="/"
    service="GOV.UK"
    serviceLink="https://www.gov.uk"
  />
  <Body
    phaseBanner={
      Object {
        "isVisible": false,
      }
    }
  />
</React.Fragment>
`;

exports[`Page layout component should render with footer when set 1`] = `
<React.Fragment>
  <Header
    logoLinkTitle=""
    menu={
      Array [
        Object {
          "label": "Logout",
          "target": "/",
        },
      ]
    }
    propositionHeader="Service Name"
    propositionHeaderLink="/"
    service="GOV.UK"
    serviceLink="https://www.gov.uk"
  />
  <Body
    phaseBanner={
      Object {
        "isVisible": false,
      }
    }
  />
  <Footer
    isVisible={true}
    links={Array []}
  />
</React.Fragment>
`;
